pages/community/community-contribute-to-precice.md

```diff
@@ -114,45 +114,15 @@ Your case may already fit into one of the existing tutorials. If not, feel free
 
-<details markdown="1"><summary>(click to read all the steps)</summary>
+Clean-up the files: remove commented-out code, remove scripts that are not needed, add case-specific files in a `.gitignore`.
 
-- Clean-up the files: remove commented-out code, remove scripts that are not needed, add case-specific files in a `.gitignore`.
-- Check your shell scripts with [shellcheck](https://github.com/koalaman/shellcheck/):
+Install `pre-commit` and enable it in the repositories you plan to contribute to with `pre-commit install`. It automatically ensures consistent formatting and best practices before you even commit changes. You can also run these checks yourself on all files using `pre-commit run -va`
 
-   ```bash
-   shellcheck <script.sh>
-   ```
-
-   and format them with any formatter (e.g. make sure there is an empty line at the end of the script).
-   Please start your shell scripts with `#!/bin/sh` and enable exit on error and undefined variables: `set -e -u`.
-- Format your `precice-config.xml` file with the [preCICE formatting tools](dev-docs-dev-tooling.html) (you need to install the [preCICE git pre-commit hook](https://github.com/precice/precice-pre-commit-hooks)):
-
-  ```bash
-  format_precice_config.py precice-config.xml
-  ```
-
-- Format your Python scripts with [PEP 8](https://pep8.org/):
-
-  ```bash
-  autopep8 --in-place --aggressive --aggressive --max-line-length 120 <file>.py
-  ```
-
-- Check your `precice-config.xml` file with the [config-visualizer](tooling-config-visualization.html). Are there any unused meshes or data?
-- Remove any comments and any explicitly-set defaults from the `precice-config.xml`. Don't worry if this sounds complicated, we will let you know in the review.
-- Check your documentation (Markdown) files with [markdownlint](https://github.com/DavidAnson/markdownlint). Install an extension for your editor, or use [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli):
-
-  ```bash
-  npm install -g markdownlint-cli
-  # See also https://stackoverflow.com/a/54170648/2254346
-  markdownlint .
-  ```
-
-We automate many of these checks with [GitHub actions](https://github.com/features/actions), which you will see running at the bottom of each pull request. To avoid pushing and waiting for the actions to run while you develop, you can alternatively install [act](https://github.com/nektos/act) to execute all or specific workflows locally, running `act` or `act -j <job_name>`. It requires [Docker](https://www.docker.com/) and you can get the latest binary from the [act releases](https://github.com/nektos/act/releases/latest).
-</details>
+We automate many checks with [GitHub actions](https://github.com/features/actions), which you will see running at the bottom of each pull request. Using `pre-commit` yourself saves you some unnecessary trouble.
```

Comment:
Do we really already include 

- checking for commented-out code
- gitignore
- shellcheck
- unused elements in the config

in the pre-commit?

Don't forget that this list also serves as a checklist for reviewers.

Reply:
"unused elements in the config" is sth we cannot do yet (you mean the preCICE config, right?), but sth we promised for preECO.

Reply:
Ah shellcheck is probably still missing. Will add that next week to the tutorials.

The gitignore and cleaning up commented code are still mentioned below. Its hard to see in all the diff noise.

Reply:
Added it.